libs/net/core/Http/OpenIdConnectRequestClient.cs

```diff
@@ -91,25 +91,27 @@ public class OpenIdConnectRequestClient : HttpRequestClient, IOpenIdConnectReque
         public async Task<string> RequestAccessToken()
         {
             HttpResponseMessage response;
-            if (_accessToken == null || String.IsNullOrWhiteSpace(_accessToken.AccessToken) || (!String.IsNullOrWhiteSpace(_accessToken.RefreshToken) && _tokenHandler.ReadJwtToken(_accessToken.RefreshToken).ValidTo <= DateTime.UtcNow))
+            var now = DateTime.UtcNow;
+            var accessToken = _accessToken?.AccessToken != null ? _tokenHandler.ReadJwtToken(_accessToken.AccessToken) : null;
+            var refreshToken = _accessToken?.RefreshToken != null ? _tokenHandler.ReadJwtToken(_accessToken.RefreshToken) : null;
+            if (accessToken == null ||
```

Comment:
This is where the bug I believe originated.  It wasn't handling the scenario where the refresh token was not provided.